rend3/src/managers/mesh.rs

```diff
@@ -229,6 +229,97 @@ impl MeshManager {
         self.registry.insert(handle, mesh);
     }
 
+    /// Duplicates a mesh's vertex data so that it can be skinned on the GPU.
+    pub fn allocate_skeleton_mesh(
+        &mut self,
+        device: &Device,
+        _queue: &Queue,
+        encoder: &mut CommandEncoder,
+        mesh_handle: &MeshHandle,
+    ) -> Range<usize> {
+        // Need to fetch internal data twice, because the returned mesh borrows &self
+        let needed_verts = self.internal_data(mesh_handle.get_raw()).vertex_range.len();
+        let vertex_range = match self.vertex_alloc.allocate_range(needed_verts) {
+            Ok(range) => range,
+            Err(_) => {
+                self.reallocate_buffers(device, encoder, needed_verts as u32, 0);
+                self.vertex_alloc
+                    .allocate_range(needed_verts)
+                    .expect("We just reallocated")
+            }
+        };
+
+        let original = self.internal_data(mesh_handle.get_raw());
+
+        encoder.copy_buffer_to_buffer(
+            &self.buffers.vertex_position,
+            original.vertex_range.start as u64,
+            &self.buffers.vertex_position,
+            vertex_range.start as u64,
+            (vertex_range.len() * VERTEX_POSITION_SIZE) as u64,
+        );
+
+        encoder.copy_buffer_to_buffer(
+            &self.buffers.vertex_normal,
+            original.vertex_range.start as u64,
+            &self.buffers.vertex_normal,
+            vertex_range.start as u64,
+            (vertex_range.len() * VERTEX_NORMAL_SIZE) as u64,
+        );
+
+        encoder.copy_buffer_to_buffer(
+            &self.buffers.vertex_tangent,
+            original.vertex_range.start as u64,
+            &self.buffers.vertex_tangent,
+            vertex_range.start as u64,
+            (vertex_range.len() * VERTEX_TANGENT_SIZE) as u64,
+        );
```

Comment:
~~We don't need to copy any of the ones the skinning shader is about to rewrite.~~

Oh we don't have a skinning shader yet... we should just flag this as to be removed.

Reply:
If we don't copy this data, when someone defines an animated mesh but forgets to add skinning to their pipeline they will get garbage. I can imagine this sort of thing happening when someone uses rend3_gltf (which will, at some point, autodetect animated meshes) but their custom rendering pipeline.

Reply:
Fair enough.